MathLib/LinAlg/Eigen/EigenOption.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef EIGENOPTION_H_
+#define EIGENOPTION_H_
+
+#include <string>
+
+namespace MathLib
+{
+
+/**
+ * \brief Option for Eigen sparse solver
+ */
+struct EigenOption
+{
+    /// Solver type
+    enum class SolverType : int
+    {
+        INVALID,
+        CG,
+        BiCGSTAB,
+        SparseLU
+    };
+
+    /// Preconditioner type
+    enum class PreconType : int
+    {
+        NONE,
+        DIAGONAL
+    };
```

Comment:
There are few more solvers available http://eigen.tuxfamily.org/dox/group__TopicSparseSystems.html
Any reasons not to include all of them in the SolverType?